otherlibs/stdune-unstable/temp.ml

```diff
@@ -103,3 +104,11 @@ let temp_path ~dir ~prefix ~suffix =
         raise Exit
       else
         candidate)
+
+let with_temp_path ~dir ~prefix ~suffix ~f =
+  match temp_path ~dir ~prefix ~suffix with
+  | exception e -> f (Error e)
+  | temp_path ->
+    let result = f (Ok temp_path) in
```

Comment:
This needs a call to protect to make sure that the temporary path is deleted even if f raises.

Reply:
Good point, will do!